### PR TITLE
Fix regex of iconsCacher

### LINKED ISSUE #10309: Svg api doesn't overwrite old image versions suffixes
- [x] Fix the regex

### PR DESCRIPTION
**Pull request by skjnldsv**
Fix #10309

### CODE REVIEW

File: lib/private/Template/IconsCacher.php [hunk 1]
```diff
@@ -47,7 +47,7 @@ class IconsCacher {
 	protected $urlGenerator;
 
 	/** @var string */
-	private $iconVarRE = '/--(icon-[a-z0-9-]+): url\(["\']([a-z0-9-_\~\/\?\&\=\.]+)[^;]+;/m';
+	private $iconVarRE = '/--(icon-[a-z0-9-]+): url\(["\']([a-z0-9-_\~\/\.]+)[^;]+;/m';
```
Comment (juliushaertl) [inline lib/private/Template/IconsCacher.php]: Just recognized it might make sense to also allow uppercase chars

Comment (skjnldsv) [reply lib/private/Template/IconsCacher.php]: Ah yes I guess :)

**Review by rullzer — APPROVED** [review]
In theory fine. But I always like it is we have test for stuff like this.

**Comment by skjnldsv** [comment]
Right! Let me add tests :)

**Comment by MorrisJobke** [comment]
Now I get this instead:

```
[Error] Failed to load resource: the server responded with a status of 404 (Not Found) (000, line 0) 
http://localhost:8000/index.php/svg/core/files/star-dark/000?v=1
```

**Comment by skjnldsv** [comment]
@MorrisJobke wrong icon, star_dark comes from core/actions. I'll fix it

**Comment by MorrisJobke** [comment]
Just noticed that it was before like this:

```
http://localhost:8000/index.php/svg/core/files/star-dark/000?v=2?v=1?v=1?v=1?v=1?v=1?v=1?v=1?v=1?v=1
```

**Review by MorrisJobke — APPROVED** [review]

**Comment by skjnldsv** [comment]
Tests added!
@nextcloud/designers please review :)

**Comment by skjnldsv** [comment]
`sh: 1: kill: No such process`
Unrelated

**Pull request merged**